base/int.jl

```diff
@@ -770,3 +770,23 @@ else
     rem(x::Int128,  y::Int128)  = checked_srem_int(x, y)
     rem(x::UInt128, y::UInt128) = checked_urem_int(x, y)
 end
+
+# bitwise ops don't need checked promotion
+for op in (:&, :|, :xor)
+    @eval function $op(a::Integer, b::Integer)
+        T = promote_typeof(a, b)
+        return $op(a % T, b % T)
+    end
+end
+
+# issue #15489: skip error checking during promotion for mixed signedness
+for op in (:+, :-, :*)
+    @eval function $op(a::Unsigned, b::Signed)
+        T = promote_typeof(a, b)
+        return $op(a % T, b % T)
+    end
+    @eval function $op(a::Signed, b::Unsigned)
+        T = promote_typeof(a, b)
+        return $op(a % T, b % T)
+    end
+end
```

Comment:
Wouldn't these definitions work for `Integer` as well? Would collapse all of these definitions.